### PR TITLE
Add scheduled queue strategy

### PR DESCRIPTION
**Pull request by rodrigo-araujo**
implements #28 

This PR creates a scheduled queue strategy.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +66,51 @@ Examples:
...
+
+    # This is optional
+
+    setup_upperkut do |config|
+      # Define which redis instance you want to use
+      config.strategy = Upperkut::Strategies::ScheduledQueue.new(
+        self,
+        redis: { url: ENV['ANOTHER_REDIS_INSTANCE_URL']) },
+        batch_size: 400, # How many events should be dispatched to worker.
+        max_wait: 300    # How long Processor wait in seconds to process batch.
```
Comment (nandosousafr) [inline README.md]: remove it :)

File: lib/upperkut/strategies/scheduled_queue.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+      def metrics
+        {
+          'latency' => latency,
+          'size' => size
+        }
+      end
+
+      def process?
+        buff_size = size('-inf', Time.zone.now.to_i)
```
Comment (nandosousafr) [inline lib/upperkut/strategies/scheduled_queue.rb]: Time.zone is a Rails method, not Ruby. Use Time.now.utc.to_i instead.

Comment (nandosousafr) [reply lib/upperkut/strategies/scheduled_queue.rb]: Let's only save events in UTC.

File: lib/upperkut/strategies/scheduled_queue.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        "upperkut:queued:#{to_underscore(@worker.name)}"
+      end
+
+      def ensure_timestamp_attr(item)
+        item['timestamp'] = Time.now.to_i unless item.key?('timestamp')
+      end
+
+      def encode_json_item(item)
+        JSON.generate(
+          'enqueued_at' => Time.now.to_i,
```
Comment (nandosousafr) [inline lib/upperkut/strategies/scheduled_queue.rb]: Time.now.utc.to_i

File: lib/upperkut/strategies/scheduled_queue.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      end
+
+      def size(min = '-inf', max = '+inf')
+        redis do |conn|
+          conn.zcount(key, min, max)
+        end
+      end
+
+      def latency
+        now = Time.now
```
Comment (nandosousafr) [inline lib/upperkut/strategies/scheduled_queue.rb]: Time.now.utc

File: lib/upperkut/strategies/scheduled_queue.rb [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          end
+        end
+
+        true
+      end
+
+      def fetch_items
+        args = {
+          value_from: '-inf'.freeze,
+          value_to: Time.now.to_f.to_s,
```
Comment (nandosousafr) [inline lib/upperkut/strategies/scheduled_queue.rb]: Time.now.utc.to_f

File: lib/upperkut/strategies/scheduled_queue.rb (hunk 2 above)
Comment (rodrigo-araujo) [reply lib/upperkut/strategies/scheduled_queue.rb]: :+1:

File: README.md [hunk 6]
```diff
@@ -66,6 +66,48 @@ Examples:
...
   ```
 
+Example 2 - Scheduled Queue:
+
+1) Create a Worker class and the define how to process the batch;
+  ```ruby
+  class MyWorker
+    include Upperkut::Worker
+
+    # This is optional
```
Comment (nandosousafr) [inline README.md]: In this case, this is not optional :)

**Comment by nandosousafr** [comment]
thanks @rodrigo-araujo

**Pull request merged**